tool/grind.dart

```diff
@@ -5,33 +5,22 @@
 import 'dart:io';
 
 import 'package:grinder/grinder.dart';
-import 'package:unscripted/unscripted.dart';
 
 import 'doc.dart';
 import 'rule.dart';
 
-main([List<String> args]) {
-  _addTask('rule',
-      parser: (String name) =>
-          generateRule(name, outDir: Directory.current.path),
-      description: 'Generate a lint rule stub.',
-      valueHelp: 'Name of rule to generate.');
+main(args) => grind(args);
 
-  _addTask('docs',
-      parser: generateDocs,
-      description: 'Generate lint rule docs.',
-      valueHelp: 'Documentation `lints/` directory.');
-
-  grind(args);
+@Task('Generate lint rule docs.')
+docs(GrinderContext context) {
+  TaskArgs args = context.invocation.arguments;
+  String dir = args.getOption('dir');
+  generateDocs(dir);
 }
 
-_addTask(String name, {String description, Parser parser, String valueHelp}) {
-  addTask(new GrinderTask(name, taskFunction: () {
-    String value = context.invocation.positionals.first;
-    parser(value);
-  },
-      description: description,
-      positionals: [new Positional(valueHelp: valueHelp)]));
+@Task('Generate a lint rule stub.')
+rule(GrinderContext context) {
+  TaskArgs args = context.invocation.arguments;
```

Comment:
I don't think `grinder` intends this to be quite so indirect; can you file a bug to improve this on the grinder side? We should at least update the docs on how to use task args. Thanks!

Reply:
(docs updated here: https://github.com/google/grinder.dart/commit/07507d4dbe7162bd9d685062f24e7a659e783eb5)

Reply:
And, you should be able to remove the explicit `GrinderContext` param to `rule()` above.

Reply:
Thanks!

Issue opened: https://github.com/google/grinder.dart/issues/317